cmd/chaininfo/chain_info.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package chaininfo
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"os"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+//go:embed arbitrum_chain_info.json
+var DefaultChainInfo []byte
+
+type ChainInfo struct {
+	ChainName             string                 `json:"chain-name"`
+	ParentChainId         uint64                 `json:"parent-chain-id"`
+	ChainParameters       *json.RawMessage       `json:"chain-parameters"`
+	ChainConfig           *params.ChainConfig    `json:"chain-config"`
+	RollupAddressesConfig *RollupAddressesConfig `json:"rollup"`
+}
+
+type RollupAddressesConfig struct {
+	Bridge                 string `koanf:"bridge"`
+	Inbox                  string `koanf:"inbox"`
+	SequencerInbox         string `koanf:"sequencer-inbox"`
+	Rollup                 string `koanf:"rollup"`
+	ValidatorUtils         string `koanf:"validator-utils"`
+	ValidatorWalletCreator string `koanf:"validator-wallet-creator"`
+	DeployedAt             uint64 `koanf:"deployed-at"`
+}
+
+func (c *RollupAddressesConfig) ParseAddresses() (RollupAddresses, error) {
+	a := RollupAddresses{
+		DeployedAt: c.DeployedAt,
+	}
+	strs := []string{
+		c.Bridge,
+		c.Inbox,
+		c.SequencerInbox,
+		c.Rollup,
+		c.ValidatorUtils,
+		c.ValidatorWalletCreator,
+	}
+	addrs := []*common.Address{
+		&a.Bridge,
+		&a.Inbox,
+		&a.SequencerInbox,
+		&a.Rollup,
+		&a.ValidatorUtils,
+		&a.ValidatorWalletCreator,
+	}
+	names := []string{
+		"Bridge",
+		"Inbox",
+		"SequencerInbox",
+		"Rollup",
+		"ValidatorUtils",
+		"ValidatorWalletCreator",
+	}
+	if len(strs) != len(addrs) {
+		return RollupAddresses{}, fmt.Errorf("internal error: attempting to parse %v strings into %v addresses", len(strs), len(addrs))
+	}
+	complete := true
+	for i, s := range strs {
+		if !common.IsHexAddress(s) {
+			log.Error("invalid address", "name", names[i], "value", s)
+			complete = false
+		}
+		*addrs[i] = common.HexToAddress(s)
+	}
+	if !complete {
+		return RollupAddresses{}, fmt.Errorf("invalid addresses")
+	}
+	return a, nil
+}
+
+func getStaticChainConfig(chainId *big.Int) (*params.ChainConfig, error) {
+	for _, potentialChainConfig := range params.ArbitrumSupportedChainConfigs {
```

Comment:
Is this still needed if we have the embedded chain info json?

Reply:
makes sense, removed.